specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-12-12/cosmos-db.json

```diff
@@ -5885,11 +5889,54 @@
           "description": "The throughput replace is pending",
           "readOnly": true
         }
+      }
+    },
+    "AutopilotSettingsResource": {
+      "type": "object",
+      "description": "Cosmos DB autopilot settings object",
+      "properties": {
+        "maxThroughput": {
+          "type": "integer",
+          "description": "Represents max throughput an autopilot container can operate at."
+        },
+        "autoUpgradePolicy": {
+          "description": "Cosmos DB resource auto-upgrade policy",
+          "$ref": "#/definitions/AutoUpgradePolicyResource"
+        },
+        "targetMaxThroughput": {
+          "type": "integer",
+          "description": "Represents target max throughput an autopilot container should operate at once offer is no longer in pending state.",
+          "readOnly": true
+        }
       },
       "required": [
-        "throughput"
+        "maxThroughput"
```

Comment:
It looks like 'throughput is now not required for 'ThroughpoutSettingsResource', is that intentional, is this no longer a required property?

Reply:
Either "throughput" is required or "autopilotSettings" settings is required. Is there a some way to convey "requiredOneOf" @markcowl

Reply:
Unfortunately, not until Swagger 3.  It is not a breakign change to remove 'required', but I would suggest documenting the fact that one of these are required in the description of the object and in the description for both parameters

Reply:
@markcowl I have updated the description of the object and in the description for both parameters